src/metabase/task/notification.clj

```diff
@@ -151,8 +152,17 @@
   Called when starting the instance."
   []
   (assert (task/scheduler) "Scheduler must be started before initializing SendPulse triggers")
-  (task/delete-all-triggers-of-job! send-notification-job-key)
-  (run! create-new-trigger! (t2/reducible-select :model/NotificationSubscription :type :notification-subscription/cron)))
+
+  ;; Get all existing triggers and subscription IDs
+  (let [existing-triggers                  (:triggers (task/job-info send-notification-job-key))
+        existing-triggers-subscription-ids (map #(get-in % [:data "subscription-id"]) existing-triggers)
+        subscription-id->cron              (t2/select-pk->fn identity :model/NotificationSubscription :type :notification-subscription/cron)
```

Comment:
I guess if having all these results in memory is too heavy, then Quartz wouldn't be coping anyway.